command/push.go

```diff
@@ -0,0 +1,301 @@
+	var archiveVCS, moduleUpload bool
+	var name string
+	args = c.Meta.process(args, false)
+	cmdFlags := c.Meta.flagSet("push")
+	cmdFlags.StringVar(&c.Meta.statePath, "state", DefaultStateFilename, "path")
+	cmdFlags.StringVar(&atlasToken, "token", "", "")
+	cmdFlags.BoolVar(&moduleUpload, "module-upload", true, "")
+	cmdFlags.StringVar(&name, "name", "", "")
+	cmdFlags.BoolVar(&archiveVCS, "vcs", true, "")
+	cmdFlags.Usage = func() { c.Ui.Error(c.Help()) }
+	if err := cmdFlags.Parse(args); err != nil {
+		return 1
+	}
+
+	// The pwd is used for the configuration path if one is not given
+	pwd, err := os.Getwd()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error getting pwd: %s", err))
+		return 1
+	}
+
+	// Get the path to the configuration depending on the args.
+	var configPath string
+	args = cmdFlags.Args()
+	if len(args) > 1 {
+		c.Ui.Error("The apply command expects at most one argument.")
+		cmdFlags.Usage()
+		return 1
+	} else if len(args) == 1 {
+		configPath = args[0]
+	} else {
+		configPath = pwd
+	}
+
+	// Verify the state is remote, we can't push without a remote state
+	s, err := c.State()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Failed to read state: %s", err))
+		return 1
+	}
+	if !s.State().IsRemote() {
+		c.Ui.Error(
+			"Remote state is not enabled. For Atlas to run Terraform\n" +
+				"for you, remote state must be used and configured. Remote\n" +
+				"state via any backend is accepted, not just Atlas. To\n" +
+				"configure remote state, use the `terraform remote config`\n" +
+				"command.")
+		return 1
+	}
+
+	// Build the context based on the arguments given
+	ctx, planned, err := c.Context(contextOpts{
+		Path:      configPath,
+		StatePath: c.Meta.statePath,
+	})
+	if err != nil {
+		c.Ui.Error(err.Error())
+		return 1
+	}
+	if planned {
+		c.Ui.Error(
+			"A plan file cannot be given as the path to the configuration.\n" +
+				"A path to a module (directory with configuration) must be given.")
+		return 1
+	}
+
+	// Get the configuration
+	config := ctx.Module().Config()
+	if name == "" {
+		if config.Atlas == nil || config.Atlas.Name == "" {
+			c.Ui.Error(
+				"The name of this Terraform configuration in Atlas must be\n" +
+					"specified within your configuration or the command-line. To\n" +
+					"set it on the command-line, use the `-name` parameter.")
+			return 1
+		}
+		name = config.Atlas.Name
+	}
+
+	// Initialize the client if it isn't given.
+	if c.client == nil {
+		// Make sure to nil out our client so our token isn't sitting around
+		defer func() { c.client = nil }()
+
+		// Initialize it to the default client, we set custom settings later
+		client := atlas.DefaultClient()
+
+		if atlasToken != "" {
+			client.Token = atlasToken
+		}
+
+		c.client = &atlasPushClient{Client: client}
+	}
+
+	// Get the variables we might already have
+	vars, err := c.client.Get(name)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf(
+			"Error looking up prior pushed configuration: %s", err))
```

Comment:
I think "previously pushed configuration" might be clearer - "prior" seems like the wrong word here.